diff -up texlive-base-20170520/source/utils/xindy/xindy-src/make-rules/inputenc/make-inp-rules.pl.perl518 texlive-base-20170520/source/utils/xindy/xindy-src/make-rules/inputenc/make-inp-rules.pl
--- texlive-base-20170520/source/utils/xindy/xindy-src/make-rules/inputenc/make-inp-rules.pl.perl518	2018-02-24 00:39:02.992893235 -0500
+++ texlive-base-20170520/source/utils/xindy/xindy-src/make-rules/inputenc/make-inp-rules.pl	2018-02-24 00:39:27.239258300 -0500
@@ -8,7 +8,7 @@ EOF
 
 
 while (<STDIN>) {
-  if (/\\indexentry{(.+)--([8-9a-f][0-9a-f])\}\{1\}/) {
+  if (/\\indexentry\{(.+)--([8-9a-f][0-9a-f])\}\{1\}/) {
     if (!($1 =~ /inputenc Error|\@inpenc\@undefined/)) {
       $i = hex($2);
       $macro = $1;
